web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -0,0 +1,341 @@
+      .enter()
+      .append('rect')
+      .attr("x", (d: BarChartData) => this.xScaleBrush(d.startTime))
+      .attr("width", this.xScaleBrush.bandwidth())
+      .attr("y", (d: BarChartData) => this.yScaleBrush(d.duration))
+      .attr("height", (d: BarChartData) => this.heightBrush - this.yScaleBrush(d.duration))
+      .attr('style', (d: BarChartData) => `fill: ${BarChartComponent.STATE_TO_COLOR[d.state]}`)
+      .attr('fill-opacity', 1)
+      .attr("stroke", function(d: BarChartData) { // Outline the white bars.
+          if (BarChartComponent.STATE_TO_COLOR[d.state] === BarChartComponent.COLOR_WHITE) {
+            return BarChartComponent.COLOR_DARK_GRAY;
+          }
+          return BarChartComponent.STATE_TO_COLOR[d.state];
+        });
+    
+    // Apply transition to all elements.
+    brushBars.selectAll('#rect')
+      .transition()
+      .duration(500);
+  }
+
+  /**
+   * This function changes the focus chart based on the input.
+   * @param inputData: Data selected by the brush or by the dropdown selection.
+   */
+  private changeFocus(inputData: BarChartData[]): void {
+    // Remove all bars of the focus chart from the previous input.
+    // TODO: Remove the boxplot points and labels.
+    if (!inputData) { return; }
+
+    const newBars = this.focus.selectAll('rect')
+      .data(inputData);
+    // TODO: Call function that creates a boxplot.
+    const maxDuration = d3.max(inputData, (d: BarChartData) => d.duration);
+    if (!maxDuration) { return; }
+
+    newBars
+      .attr('class', '.newBars')
+      .enter()
+      .append('rect') 
+        .attr('x', (d: BarChartData) => this.xScaleFocus(d.startTime))
+        .attr('width', this.xScaleFocus.bandwidth())
+        .attr('y', (d: BarChartData) => this.yScaleFocus(d.duration))
+        .attr('height', (d: BarChartData) => 
+          this.yScaleFocus(0) - this.yScaleFocus(d.duration))
+        .attr('style', (d: BarChartData) => 
+          `fill: ${BarChartComponent.STATE_TO_COLOR[d.state]}`)
+        .attr('fill-opacity', 1)
+        .attr("stroke", function(d: BarChartData) { // Outline the white bars.
+            if (BarChartComponent.STATE_TO_COLOR[d.state] === BarChartComponent.COLOR_WHITE) {
+              return BarChartComponent.COLOR_DARK_GRAY;
+            }
+            return BarChartComponent.STATE_TO_COLOR[d.state];
+          });
+      // TODO: Call mouseover and mouseleave function here.
+
+    // Add transparent bars for hover convience.
+    newBars
+      .attr('class', '.transBars')
+      .enter()
+      .append('rect') // Add a transparent rect for each element.
+        .attr('x', (d: BarChartData) => this.xScaleFocus(d.startTime))
+        .attr('width', this.xScaleFocus.bandwidth())
+        .attr('y', (d: BarChartData) => this.yScaleFocus(maxDuration))
+        .attr('height', (d: BarChartData) => 
+          this.yScaleFocus(d.duration) - this.yScaleFocus(maxDuration))
+        .attr('fill', BarChartComponent.COLOR_WHITE_TRANS);
```

Comment:
Any reason why the indent shifts here?

Reply:
The indented line belongs to the previous one, that's why it is differently indented.